src/lizaalert/authentication/views.py

```diff
@@ -111,3 +123,46 @@ def post(self, request, *args, **kwargs):
         if request.user and request.user.username == "test_user":
             return Response(status=status.HTTP_200_OK)
         return Response(status=status.HTTP_403_FORBIDDEN)
+
+
+class TokenExchange(APIView):
+    """
+    Замена OAuth-токена Яндекс ID на JWT-токен.
+
+    Методы:
+    - POST: Принимает OAuth-токен Яндекс, Возвращает acsess и refresh JWT-токены.
+
+    """
+
+    permission_classes = [AllowAny]
+
+    @swagger_auto_schema(
+        operation_description="Принимает OAuth-токен Яндекс, Возвращает acsess и refresh JWT-токены",
+        request_body=OauthTokenSerializer,
+        responses={
+            201: TokenRefreshSerializer,
+            400: Error400Serializer,
+            401: YandexResponseStatusSerializer,
+        },
+    )
+    def post(self, request):
+        yandex_user_data, status_code = self.get_yandex_user_data(request.data["oauth_token"])
+        if not yandex_user_data:
+            serializer = YandexResponseStatusSerializer(data={"yandex_response_status": status_code})
+            if serializer.is_valid():
+                return Response(serializer.data, status=status.HTTP_401_UNAUTHORIZED)
+            return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+        user, _ = User.objects.get_or_create(id=int(yandex_user_data.id), username=yandex_user_data.login)
+        refresh = RefreshToken.for_user(user)
+        data = {"refresh": str(refresh), "access": str(refresh.access_token)}
+        return Response(data, status=status.HTTP_201_CREATED)
+
+    @staticmethod
+    def get_yandex_user_data(oauth_token):
+        headers = {"Authorization": f"OAuth {oauth_token}"}
+        request = requests.get(settings.YANDEX_INFO_URL, headers=headers)
+        if request.status_code == status.HTTP_200_OK:
+            user_data = namedtuple("UserData", ("id", "login"))(request.json()["id"], request.json()["login"])
```

Comment:
нужно один раз прочитать json
а определения тапла можно вынести из функции

```suggestion
            req_data = request.json()
            user_data = UserData(req_data["id"], req_data["login"])
```